lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain_record.py

```diff
@@ -0,0 +1,492 @@
+        self.payload = self.__build_payload()
+        self.force_update = module.params.get('force_update', False)
+        self.record_id = module.params.get('record_id', None)
+
+    def check_credentials(self):
+        # Check if oauth_token is valid or not
+        response = self.get('account')
+        if response.status_code == 401:
+            self.module.fail_json(msg='Failed to login using oauth_token, please verify validity of oauth_token')
+
+    def verify_domain(self):
+        # URL https://api.digitalocean.com/v2/domains/[NAME]
+        response = self.get('domains/%s' % self.domain)
+        status_code = response.status_code
+        json = response.json
+
+        if status_code not in (200, 404):
+            self.module.fail_json(msg='Error getting domain [%(status_code)s: %(json)s]' % {'status_code': status_code, 'json': json})
+        elif status_code == 404:
+            self.module.fail_json(msg="No domain named '%s' found. Please create a domain first" % self.domain)
+
+    def __get_all_records(self):
+
+        records = []
+        page = 1
+        while True:
+            # GET /v2/domains/$DOMAIN_NAME/records
+            response = self.get('domains/%(domain)s/records?page=%(page)s' % {'domain': self.domain, 'page': page})
+            status_code = response.status_code
+            json = response.json
+
+            if status_code != 200:
+                self.module.fail_json(msg='Error getting domain records [%(status_code)s: %(json)s]' % {'status_code': status_code, 'json': json})
+
+            for record in json['domain_records']:
+                records.append(dict([(str(k), v) for k, v in record.items()]))
+
+            if 'pages' in json['links'] and 'next' in json['links']['pages']:
+                page += 1
+            else:
+                break
+
+        return records
+
+    def __normalize_data(self):
+        # for the MX, CNAME, SRV, CAA records make sure the data ends with a dot
+        if (self.payload['type'] in ['CNAME', 'MX', 'SRV', 'CAA'] and self.payload['data'] != '@' and not self.payload['data'].endswith(".")):
+            data = "%s." % self.payload['data']
+        else:
+            data = self.payload['data']
+
+        return data
+
+    def __find_record_by_id(self, record_id):
+        for record in self.records:
+            if record['id'] == record_id:
+                return record
+        return None
+
+    def __get_matching_records(self):
+        """Collect exact and similar records
+
+        It returns an exact record if there is any match along with the record_id.
+        It also returns multiple records if there is no exact match
+        """
+
+        # look for exactly the same record used by (create, delete)
+        for record in self.records:
+            r = dict(record)
+            del r['id']
+            # python3 does not have cmp so let's use the official workaround
+            if (r > self.payload) - (r < self.payload) == 0:
```

Comment:
This blows up for me with:
```
/home/phemmer/.ansible/tmp/ansible-tmp-1547002269.7-214143530475396/AnsiballZ_digital_ocean_domain_record.py:17: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "/home/phemmer/.ansible/tmp/ansible-tmp-1547002269.7-214143530475396/AnsiballZ_digital_ocean_domain_record.py", line 113, in <module>
    _ansiballz_main()
  File "/home/phemmer/.ansible/tmp/ansible-tmp-1547002269.7-214143530475396/AnsiballZ_digital_ocean_domain_record.py", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/phemmer/.ansible/tmp/ansible-tmp-1547002269.7-214143530475396/AnsiballZ_digital_ocean_domain_record.py", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/usr/lib64/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib64/python3.7/imp.py", line 169, in load_source
    module = _exec(spec, sys.modules[name])
  File "<frozen importlib._bootstrap>", line 630, in _exec
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/ansible_digital_ocean_domain_record_payload_a66bc9an/__main__.py", line 492, in <module>
  File "/tmp/ansible_digital_ocean_domain_record_payload_a66bc9an/__main__.py", line 484, in main
  File "/tmp/ansible_digital_ocean_domain_record_payload_a66bc9an/__main__.py", line 334, in create_or_update_record
  File "/tmp/ansible_digital_ocean_domain_record_payload_a66bc9an/__main__.py", line 296, in __get_matching_records
TypeError: '>' not supported between instances of 'dict' and 'dict'
```

Reply:
@phemmer This code hasn't been updated for 1.5 years. :( So the python 3.7 specific dict compare was not written / updated. In the meantime I had to realize that ansible is definitely not the right tool to manage domain records/entries so I'm using `doctl` and `terraform` to maintain the desired state instead. Although I used my own plugin to create tons of records 1.5 years ago.

Reply:
I was able to fix this locally by changing to `if r == self.payload:`. No idea if there's some case this doesn't work for though.

At scale ansible might not be the right tool. However I have no desire to deal with adding another tool just for updating a single record.